packages/@pollyjs/adapter-node-http/src/-private/transport-wrapper.js

```diff
@@ -0,0 +1,292 @@
+
+    transport.request = request;
+    transport.get = get;
+
+    nativeRequestMapping.delete(transport);
+  }
+
+  getBodyFromChunks(chunks, headers) {
+    const { adapter } = this;
+
+    // If content-encoding is set in the header then the body/content
+    // should not be concatenated. Instead, the chunks should
+    // be preserved as-is so that each chunk can be mocked individually
+    if (isContentEncoded(headers)) {
+      const hexChunks = chunks.map(chunk => {
+        if (!Buffer.isBuffer(chunk)) {
+          adapter.assert(
+            'content-encoded responses must all be binary buffers',
+            typeof chunk === 'string'
+          );
+          chunk = Buffer.from(chunk);
+        }
+
+        return chunk.toString('hex');
+      });
+
+      return JSON.stringify(hexChunks);
+    }
+
+    const buffer = mergeChunks(chunks);
+
+    // The merged buffer can be one of two things:
+    //  1. A binary buffer which then has to be recorded as a hex string.
+    //  2. A string buffer.
+    return buffer.toString(isBinaryBuffer(buffer) ? 'hex' : 'utf8');
+  }
+
+  getChunksFromBody(body, headers) {
+    if (!body) {
+      return [];
+    }
+
+    if (Buffer.isBuffer(body)) {
+      return [body];
+    }
+
+    // If content-encoding is set in the header then the body/content
+    // is as an array of hex strings
+    if (isContentEncoded(headers)) {
+      const hexChunks = JSON.parse(body);
+
+      return hexChunks.map(chunk => Buffer.from(chunk, 'hex'));
+    }
+
+    const buffer = Buffer.from(body);
+
+    // The body can be one of two things:
+    //  1. A hex string which then means its binary data.
+    //  2. A utf8 string which means a regular string.
+    return [Buffer.from(buffer, isBinaryBuffer(buffer) ? 'hex' : 'utf8')];
+  }
+
+  async passthrough(pollyRequest) {
+    const { transport } = this;
+    const [, , args] = pollyRequest.requestArguments;
+    const { method, headers, body: chunks } = pollyRequest;
+    const { request: nativeRequest } = nativeRequestMapping.get(transport);
+    let [url, options] = args;
+
+    /**
+     * args could be (url, options, callback) or (options, callback) depending
+     * on nodejs version. If the `url` arguments is not a string or URL
+     * instance, then use (options, callback).
+     */
+    if (isObjectLike(url) && !url.searchParams) {
+      options = url;
+      url = undefined;
+    }
+
+    const request = nativeRequest.call(transport, {
+      ...options,
+      method,
+      headers: { ...headers },
+      ...NodeUrl.parse(pollyRequest.url)
+    });
+
+    // Write the request body
+    chunks.forEach(chunk => request.write(chunk));
+
+    const response = await new Promise(resolve => {
+      request.once('response', response => resolve(response));
+      request.end();
+    });
+
+    const responseBody = await new Promise(resolve => {
+      const chunks = [];
+
+      response.on('data', chunk => chunks.push(chunk));
```

Comment:
Should we need to handle the `error` event?

Reply:
Done 😄